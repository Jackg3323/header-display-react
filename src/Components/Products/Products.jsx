import PropTypes from "prop-types";

export default function Products({ products }) {
  const listItems = products.map((product) => (
    <li key={product.name}>
      {product.name} {product.price}
    </li>
  ));
  return <ul>{listItems}</ul>;
}

Products.propTypes = {
  products: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      price: PropTypes.string.isRequired,
    })
  ).isRequired,
};
