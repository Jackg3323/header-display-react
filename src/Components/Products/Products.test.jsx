import renderer from "react-test-renderer";
import Products from "./Products";

const products = [
  {
    price: "$49.99",
    name: "Test 1",
  },
  {
    price: "$9.99",
    name: "Test 2",
  },
  {
    price: "$29.99",
    name: "Test 3",
  },
];

it("renders without crashing", () => {
  const rendered = renderer.create(<Products products={products} />).toJSON();
  expect(rendered).toMatchSnapshot();
});
