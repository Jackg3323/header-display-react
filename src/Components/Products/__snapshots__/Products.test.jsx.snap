// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<ul>
  <li>
    Test 1
     
    $49.99
  </li>
  <li>
    Test 2
     
    $9.99
  </li>
  <li>
    Test 3
     
    $29.99
  </li>
</ul>
`;
